fusion.md

```diff
@@ -0,0 +1,1396 @@
+    // scale heating and cooling degree days for the next period
+    mCurrDDScaler = 1.0 / ( currGlobalEmiss / mBaseYearValue ) * mHDDCoef;
+```
+
+Finally we can query for the appropriate GCAM paramaters again but this time
+changing the value.  You will notice that really everything works the same as
+when we were collecting the CO2 emissions.  The data passed to `processData` is
+passed by reference to the actual parameter that lives in the GCAM objects and
+is not const so we are free to change it as we please.  These are the queries
+for building heating and cooling services:
+
+```cpp
+    // Note the actual services are "resid heating" or "comm cooling", etc so we
+    // use regular expression partial matching so we do not have to spell it out.
+    vector<FilterStep*> ddFilterSteps = parseFilterString( "world/region/consumer/nodeInput/nodeInput/nodeInput[NamedFilter,StringRegexMatches,heating]" );
+    ddFilterSteps.push_back( new FilterStep( "degree-days", new IndexFilter( new IntEquals( aPeriod + 1 ) ) ) );
+    GCAMFusion<DegreeDaysFeedback> scaleHDD( *this, ddFilterSteps );
+    scaleHDD.startFilter( aScenario );
+    
+    mCurrDDScaler = ( currGlobalEmiss / mBaseYearValue ) * mCDDCoef;
+    // only updating the service name filter of our query, we can keep the rest of it the same
+    delete ddFilterSteps[ ddFilterSteps.size() - 2 ];
+    ddFilterSteps[ ddFilterSteps.size() - 2 ] = new FilterStep( "nodeInput", new NamedFilter( new StringRegexMatches( "cooling" ) ) );
+    GCAMFusion<DegreeDaysFeedback, true> scaleCDD( *this, ddFilterSteps );
+    scaleCDD.startFilter( aScenario );
+
+    // note we are still responsible for the memory we allocate even if it was done in the parseFilterString utility
+    for( auto filterStep : ddFilterSteps ) {
+        delete filterStep;
+    }
+```
+
+And here are the call backs that set the scaled degree days in those sectors:
+
+```cpp
+template<typename T>
+void DegreeDaysFeedback::processData( T& aData ) {
+    assert( false );
+}
+template<>
+void DegreeDaysFeedback::processData<Value>( Value& aData ) {
+    // We are manipulating aData which is referenced back to the actual GCAM objects
+    aData *= mCurrDDScaler;
+}
+
+void DegreeDaysFeedback::pushFilterStep( INamed* const& aContainer ) {
+    std::cout << "Saw step " << aContainer->getName() << std::endl;
+}
+
+template<typename T>
+typename boost::disable_if<
+    boost::is_base_of<INamed, typename boost::remove_pointer<T>::type>,
+void>::type DegreeDaysFeedback::pushFilterStep( const T& aContainer ) {
+    std::cout << "Saw unknown " << typeid( T ).name() << std::endl;
+}
+```
+
+A couple things to note, sometimes it is easier to just use your feedback object to process callbacks from GCAM fusion.  It isn't required to use a helper struct to do so and sometimes it is easier not to.  Nothing additional is required to handle feedbacks, you don't have to implement any interface, just provide the `processData`, etc callback methods.
```

Comment:
We use just about every conceivable variant of "GCAMFusion", "GCAM Fusion", "GCAM fusion", etc.  Obviously there is going to be some variation because some of the uses are C++ identifiers and have to conform to those requirements.  For general usage in text we should pick one variant and stick with it.

Reply:
I'll go back through with some regular expressions and just settle on "GCAM Fusion" (as appropriate).